CDCgov/NEDSS-Modernization
# Cnft2-831 Pagebuilder DEV deployment prep

## Description

Prior to deployment of the pagebuilder-api, the base path needs to be update to `/nbs/page-builder` and a configuration endpoint needs to be created to allow enabling/disabling page builder for specific environments.

## Tickets

* [CNFT2-831](https://cdc-nbs.atlassian.net/browse/CNFT2-831)


## Steps to verify

1. Start page-builder api in docker `docker-compose up pagebuilder-api --build -d`
1. Rebuild modernization-api (ui) `docker-compose up modernization-api --build -d`
1. Delete existing Traefik container
1. Start Traefik with updated route (in cdc-sandbox) `docker-compose up reverse-proxy -d`
1. Ensure all containers are running in docker
1. Navigate to [NBS](http://localhost:8080/nbs/login)
1. Log in as `superuser`
2. Navigate to [Page builder](http://localhost:8080/page-builder/manage/pages)
3. Verify the page loads
4. Re-start `modernization-api` after updating the `docker-compose` to disable page builder: 
```
 - NBS_UI_FEATURES_PAGEBUILDER_ENABLED=false
```
Alternatively you can run the applications outside of docker and update the pagebuilder config by running the modernization api as follows:
```sh
./gradlew :modernization-api:bootRun --args='--server.port=9080 --nbs.ui.features.pageBuilder.enabled=false'
```

11. Attempt to navigate to [Page builder](http://localhost:8080/page-builder/manage/pages)
12. Verify re-routing to advanced search





[CNFT2-831]: https://cdc-nbs.atlassian.net/browse/CNFT2-831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
